community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/rewriters/NormalizeWithClausesTest.scala

```diff
@@ -223,6 +225,34 @@ class NormalizeWithClausesTest extends CypherFunSuite with RewriteTest {
     )
   }
 
+  test("rejects use of aggregation in ORDER BY if aggregation is not used in associated WITH") {
+    // Note: aggregations in ORDER BY that don't also appear in WITH are invalid
+    try {
```

Comment:
In ScalaTest we have intercept, which makes this type of testing a bit nicer. An example of how to use it here:
https://github.com/neo4j/neo4j/blob/2.3/community/cypher/cypher/src/test/scala/org/neo4j/cypher/ErrorMessagesTest.scala#L433